engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -739,6 +740,86 @@ namespace dmGameSystem
+        minPoint = Vector3(data[0], data[1], data[2]);
+        maxPoint = Vector3(data[3], data[4], data[5]);
+    }
+
+    static void RenderListFrustumCulling(dmRender::RenderListVisibilityParams const &params)
+    {
+        DM_PROFILE("Mesh");
+
+        MeshWorld* mesh_world = (MeshWorld*)params.m_UserData;
+
+
+        const dmIntersection::Frustum frustum = *params.m_Frustum;
+        uint32_t num_entries = params.m_NumEntries;
+        for (uint32_t i = 0; i < num_entries; ++i)
+        {
+            dmRender::RenderListEntry* entry = &params.m_Entries[i];
+            MeshComponent* component_p = (MeshComponent*)entry->m_UserData;
+
+            dmGameSystem::BufferResource* br = GetVerticesBuffer(component_p, component_p->m_Resource);
+            dmVMath::Vector3 boundsMin, boundsMax;
+
+            void* data;
+            uint32_t count;
+            dmBuffer::ValueType valueType;
+            dmBuffer::Result r = dmBuffer::GetMetaData(br->m_Buffer, dmHashString64("AABB"), &data, &count, &valueType );
+            if (r == dmBuffer::RESULT_METADATA_MISSING) {
+                // no bounding box available - no culling
+                entry->m_Visibility = dmRender::VISIBILITY_FULL; // TODO - or shall we just "continue"
+                continue;
+            } else
+            if (r != dmBuffer::RESULT_OK) {
+                dmLogError("Error getting AABB for mesh buffer");
+                continue;
+            }
+
+            switch (valueType) {
+                case dmBuffer::VALUE_TYPE_FLOAT32:
+                    MetadataAabbToPoints((float*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_UINT8:
+                    MetadataAabbToPoints((uint8_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_UINT16:
+                    MetadataAabbToPoints((uint16_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_UINT32:
+                    MetadataAabbToPoints((uint32_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_UINT64:
+                    MetadataAabbToPoints((uint64_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_INT8:
+                    MetadataAabbToPoints((int8_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_INT16:
+                    MetadataAabbToPoints((int16_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_INT32:
+                    MetadataAabbToPoints((int32_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_INT64:
+                    MetadataAabbToPoints((int64_t*)data, valueType, boundsMin, boundsMax);
```

Comment:
I think the extra code generated for something that we know the resulting data type for is unnessary.

E.g. this should work fine:
```
Vector3 bmin( (float)aabb[0], (float)aabb[1], (float)aabb[2] );
Vector3 bmax( (float)aabb[3], (float)aabb[4], (float)aabb[5] );
```

Reply:
No, but that's the "API", and what we expect, so we only need to support that.

Reply:
ok